src/Microsoft.IdentityModel.Tokens/X509SecurityKey.cs

```diff
@@ -127,6 +127,26 @@ public override SignatureProvider GetSignatureProvider(string algorithm, bool ve
                 return CryptoProviderFactory.CreateForSigning(this, algorithm);
         }
 
+        public override bool IsSupportedAlgorithm(string algorithm)
+        {
+            if (string.IsNullOrEmpty(algorithm))
+                return false;
+
+            switch (algorithm)
+            {
+                case SecurityAlgorithms.RsaSha256:
+                case SecurityAlgorithms.RsaSha384:
+                case SecurityAlgorithms.RsaSha512:
+                case SecurityAlgorithms.RsaSha256Signature:
+                case SecurityAlgorithms.RsaSha384Signature:
+                case SecurityAlgorithms.RsaSha512Signature:
+                    return _certificate.PublicKey != null;
```

Comment:
Heh, this check doesn't seem rock-solid. It will return a wrong value for DSA/ECDsa certificates.